packages/poseidon-proof/tests/index.test.ts

```diff
@@ -43,48 +43,46 @@ const poseidonFunctions = [
 
 const computePoseidon = (preimages: string[]) => poseidonFunctions[preimages.length - 1](preimages)
 
-describe("PoseidonProof", () => {
-    const preimages = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]
-    const scope = 1
+const preimages = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16]
+const scope = 1
+let curve: any
+const proofs: Array<{ fullProof: PoseidonProof; digest: bigint }> = []
 
-    let fullProof: PoseidonProof
-    let curve: any
+beforeAll(async () => {
+    curve = await buildBn128()
+    for (const preimage of preimages) {
+        const currentPreimages = preimages.slice(0, preimage)
+        const fullProof = await generate(currentPreimages, scope)
```

Comment:
Do we really need to test all the instances here?

Reply:
Well each involves different artifacts. So I'd say yes?

Reply:
Right, just wondering how much time it needs.

Reply:
timeout was already set at 1min. So I kept that value